src/sentry/api/endpoints/relocation.py

```diff
@@ -0,0 +1,122 @@
+            return Response({"detail": ERR_DUPLICATE_RELOCATION}, status=400)
+
+        serializer = RelocationPostSerializer(data=request.data)
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=400)
+
+        validated = serializer.validated_data
+        fileobj = validated.get("file")
+        owner_username = validated.get("owner")
+        org_slugs = validated.get("orgs")
+        try:
+            owner = user_service.get_by_username(username=owner_username)[0]
+        except IndexError:
+            return Response({"detail": f"Could not find user `{owner_username}`"}, status=400)
+
+        # Quickly check that this `owner` does not have more than one active `Relocation` in flight.
+        if Relocation.objects.filter(
+            owner=owner.id, status=Relocation.Status.IN_PROGRESS.value
+        ).exists():
+            return Response({"detail": ERR_DUPLICATE_RELOCATION}, status=409)
+
+        # TODO(getsentry/team-ospo#203): check import size, and maybe do throttle based on that
+        # information.
+
+        file = File.objects.create(name="raw-relocation-data.tar", type="relocation.file")
+        file.putfile(fileobj, blob_size=RELOCATION_BLOB_SIZE, logger=logger)
+
+        with atomic_transaction(
+            using=(router.db_for_write(Relocation), router.db_for_write(RelocationFile))
+        ):
+            relocation: Relocation = Relocation.objects.create(
+                creator=request.user.id,
+                owner=owner.id,
+                want_org_slugs=org_slugs,
+                step=Relocation.Step.UPLOADING.value,
+            )
+            RelocationFile.objects.create(
+                relocation=relocation,
+                file=file,
+                kind=RelocationFile.Kind.RAW_USER_DATA.value,
+            )
+
+        return Response(status=201)
+
+
+@region_silo_endpoint
+class RelocationEndpoint(Endpoint, RelocationMixin):
+    owner = ApiOwner.RELOCATION
+    publish_status = {
+        "POST": ApiPublishStatus.EXPERIMENTAL,
+    }
+    permission_classes = (SuperuserPermission,)
+
+    def post(self, request: Request) -> Response:
+        """
+        Upload an encrypted export tarball for relocation.
+        ``````````````````````````````````````````````````
+
+        Upload an encrypted relocation tarball for relocation.
+
+        This is currently an experimental API, and for the time being is only meant to be called by
+        admins.
+
+        :param file file: the multipart encoded tarball file.
+        :param string owner: the username of the "owner" of this relocation; not necessarily
+                             identical to the user who made the API call.
+        :param list[string] orgs: A list of org slugs from those included in the associated
+                                  encrypted backup tarball that should be imported.
+        :auth: required
+        """
+
+        logger = logging.getLogger(__name__)
```

Comment:
this is already assigned at the module level

Reply:
Removed.